04-kotlin-socket/src/main/kotlin/SocketNetworkManager.kt

```
@@ -0,0 +1,62 @@
+package demo
+
+import org.protelis.lang.datatype.DeviceUID
+import org.protelis.vm.CodePath
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.net.ServerSocket
+import java.net.Socket
+import java.net.SocketTimeoutException
+import kotlin.concurrent.thread
+
+class SocketNetworkManager(private val uid: DeviceUID, private val port: Int, private val neighbors: Set<IPv4Host>) : MyNetworkManager {
+    private val timeout = 1000
+    private var toBeSent: Map<CodePath, Any> = emptyMap()
+    private var messages: Map<DeviceUID, Map<CodePath, Any>> = emptyMap()
+    private var running = false
+
+    fun listen() {
+        running = true
+        val server = ServerSocket(port).also { it.soTimeout = timeout }
+        thread {
+            while (running) {
+                try {
+                    handleConnection(server.accept())
+                } catch (e: SocketTimeoutException) { }
```

Never silence exceptions

---

Prefer NIO async channels: https://www.baeldung.com/java-nio2-async-socket-channel